scipy/optimize/tests/test_bracket.py

```diff
@@ -157,76 +166,83 @@ def f(*args, **kwargs):
             i = rng.random(size=shape) > 0.5
             xmin[i], xmax[i] = -np.inf, np.inf
         factor = rng.random(size=shape) + 1.5
+        refs = xp.asarray(bracket_root_single(xl0, xr0, xmin, xmax, factor, p).ravel())
```

Comment:
CuPy chokes here because xp.asarray is fed something with `dtype=object` and CuPy complains `ValueError: Unsupported dtype object`.

`torch` is similar with `TypeError: can't convert np.ndarray of type numpy.object_`

we may need to skip for those backends for now? Looks like the dtype is indeed always `object`, even with NumPy case.

I *think* they are allowed to refuse `object` because of https://data-apis.org/array-api/latest/API_specification/data_types.html#data-types I guess.

Reply:
Ah yes because this is an array of results objects, I should just call `xp.asarray` on the attributes when required